src/librustc/ty/sty.rs

```diff
@@ -614,6 +614,14 @@ impl<'tcx> TraitRef<'tcx> {
         TraitRef { def_id: def_id, substs: substs }
     }
 
+    /// Returns a TraitRef of the form `Self: Foo<U>`.
+    pub fn identity<'a, 'gcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>, def_id: DefId) -> TraitRef<'tcx> {
+        TraitRef {
+            def_id,
+            substs: Substs::identity_for_item(tcx, def_id),
```

Comment:
Can you make sure no existing code uses `identity_for_item` with `TraitRef`?

Reply:
yeah

Reply:
found a few uses, commit incoming